src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/dto/utils/MaybeChanged.kt

```diff
@@ -0,0 +1,24 @@
+package uk.gov.justice.digital.hmpps.incidentreporting.dto.utils
+
+/**
+ * Represents a value that might have been changed,
+ * allowing a block to be called only if it had been changed.
+ */
+sealed interface MaybeChanged<T> {
+  val value: T
+
+  fun alsoIfChanged(block: (T) -> Unit): MaybeChanged<T>
+
+  data class Unchanged<T>(override val value: T) : MaybeChanged<T> {
+    override fun alsoIfChanged(block: (T) -> Unit): Unchanged<T> {
+      return this
```

Comment:
Isn't it worth calling the block anyway? Maybe the client wants to do something on the unchanged thing? 🤔

Reply:
I see, this method is just here so that client can call this regardless of the value but it would only invoke the block if stuff actually changed 👌